### ISSUE TITLE
significance tests

### ISSUE THREAD

**Issue by matt-long** [NCAR/esmlab#89]
For our statistics functions, like `weighted_cor` it would be nice to enable computing significance metrics (p-value...).

@bradyrx, perhaps you can help with this?

**Comment by bradyrx** [comment]
@matt-long, yes I'd be happy to open a PR on this. Are you trying to keep `esmlab` slim on dependencies? 

`xskillscore` has a pearson_r_p_value function that is based in numpy and written as an `xarray` ufunc. Although this only allows a two-tailed test.

I can also port over some functions from `climpred` (https://github.com/bradyrx/climpred/blob/master/climpred/stats.py) which accounts for autocorrelation in the time series by computing the effective sample size (Bretherton et al. 1999).

**Comment by matt-long** [comment]
I am fine with dependencies as long as they are good ones.

**Comment by bradyrx** [comment]
Implemented in https://github.com/NCAR/esmlab/pull/90, although should stay open until effective sample size is incorporated (will get on this when time allows).
